crates/control/src/behavior/walk_to_pose.rs

```diff
@@ -54,6 +56,20 @@ impl<'cycle> WalkPathPlanner<'cycle> {
             );
         }
 
+        if let Some(GameControllerState {
+            game_state: GameState::Playing,
+            sub_state: Some(SubState::PenaltyKick),
+            kicking_team,
+            ..
+        }) = game_controller_state
+        {
+            planner.with_penalty_box(
+                robot_to_field.inverse(),
+                self.field_dimensions,
+                kicking_team,
+            )
+        };
+
```

Comment:
What do you think about moving this higher level behavior decision to outside this call. Something like `should_avoid_penalty_box`?

Reply:
has been changed as discussed